### PR TITLE
feat: Add batch float32 zero-copy angle encoding in qdp-core

### LINKED ISSUE #1267: [Feature] Add batch float32 zero-copy angle encoding in qdp-core
Reference: #1004 

### What

Add batch float32 support for angle encoding in QDP core and CUDA kernels.

This includes:
- host-side `encode_batch_f32(..., "angle")` support
- zero-copy CUDA pointer batch APIs for angle float32 input
- CUDA kernel and FFI support for batched float32 angle encoding
- validation and tests for invalid shapes, null pointers, zero samples, and non-finite inputs

### Why

QDP already supports angle float32 zero-copy encoding for single samples, but batch angle workloads still do not have the same float32 core support.

That leaves a gap for the common case where training or inference pipelines already hold angle features as batched CUDA float32 tensors. Adding batch float32 support:
- keeps angle float32 behavior consistent between single-sample and batch paths
- avoids unnecessary dtype promotion for batched angle inputs
- preserves zero-copy integration for framework-owned CUDA tensors
- aligns the angle implementation with the existing amplitude batch float32 pattern

### PR DESCRIPTION
**Pull request by viiccwen**
### Related Issues

- Closes #1267

### Changes

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation
- [x] Test
- [ ] CI/CD pipeline
- [ ] Other

---

- add batched float32 CUDA kernel support for angle encoding
- expose the new kernel through `qdp-kernels` FFI
- implement host-side angle batch float32 encoding in `qdp-core`
- add zero-copy batch float32 angle APIs for CUDA pointers, including stream-aware variants
- align float32 GPU-pointer validation with the existing float64 angle batch behavior
- add kernel and core tests for success paths and failure paths

### Why

QDP already had angle float32 zero-copy support for single samples, but not for batched angle features. That made batched angle workloads inconsistent with the existing single-sample API and with the amplitude float32 batch implementation pattern.

This change closes that gap in the Rust/CUDA core only. Python bindings remain out of scope for a follow-up PR.

### How

- implement `launch_angle_encode_batch_f32` in `qdp-kernels/src/angle.cu`
- add the corresponding extern declaration and non-CUDA stub in `qdp-kernels/src/lib.rs`
- extend `AngleEncoder` with host-side batch float32 encoding and batch GPU-pointer float32 encoding
- add public `QdpEngine` entry points for angle batch float32 zero-copy on the default stream and explicit streams
- validate sample shape and reject non-finite float32 GPU-resident batch inputs before launching the encoding kernel
- add tests covering kernel correctness, host-side batch encoding, zero-copy batch encoding, stream interop, error handling, and Float64-engine conversion

## Checklist

- [x] Added or updated unit tests for all changes
- [ ] Added or updated documentation for all changes

### CODE REVIEW

File: qdp/qdp-core/src/lib.rs [hunk 1]
```diff
@@ -768,6 +768,78 @@ impl QdpEngine {
         Ok(batch_state_vector.to_dlpack())
     }
 
+    /// Encode an angle batch from an existing GPU pointer (float32 input only).
+    ///
+    /// Zero-copy batch encoding from CUDA float32 tensors. Uses the default CUDA stream.
+    /// For stream interop use `encode_angle_batch_from_gpu_ptr_f32_with_stream`.
```
Comment (rich7420) [inline qdp/qdp-core/src/lib.rs:774]: nit: doc should mention that output precision follows the engine's configured precision, not necessarily f32 — F64 engine + F32 input yields complex128. Don't want downstream folks tripping on the DLPack dtype.

**Review by rich7420 — APPROVED** [review]
@viiccwen thanks for the patch!
Could you add a NaN case to mirror the Infinity one, and also state_len == 0 /num_qubits == 0 launcher-rejection tests?

**Comment by rich7420** [comment]
@viiccwen thanks for the patch!
